### PR TITLE
Cleaned, rebased and squashed commits for machine-id

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java [hunk 1]
```diff
@@ -435,6 +436,25 @@ private static void determineMetricSetPrometheus(ModelNode config,
...
+            throws OperationFailedException {
+
+        if (!config.hasDefined(PlatformDefinition.PLATFORM)) {
+            return null; // not monitoring platform, so we don't collect machine ID
+        }
+
+        List<Property> asPropertyList = config.get(PlatformDefinition.PLATFORM).asPropertyList();
+        if (asPropertyList.size() == 0) {
+            return null; // not monitoring platform, so we don't collect machine ID
+        } else if (asPropertyList.size() > 1) {
```
Comment (abonas) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:449]: if only one is allowed, why line 446 works with a list?

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java [hunk 2]
```diff
@@ -496,7 +528,7 @@ private static void determineMetricSetPrometheus(ModelNode config,
 
                     MetricType<PlatformNodeLocation> usableSpace = new MetricType<PlatformNodeLocation>(null,
                             Constants.FILE_STORE_USABLE_SPACE,
-                            new AttributeLocation<PlatformNodeLocation>(
+                            new AttributeLocation<>(
```
Comment (abonas) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:531]: why is this change related to machine id? (and a similar one below)

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/util/Util.java [hunk 3]
```diff
@@ -301,4 +309,29 @@ public static MonitorServiceConfiguration getMonitorServiceConfiguration(Operati
...
+     */
+    public static String getSystemId() {
+        if (systemId == null) {
+            File machineIdFile = new File("/etc/machine-id");
+            if (machineIdFile.exists() && machineIdFile.canRead()) {
+                try (Reader reader = new InputStreamReader(new FileInputStream(machineIdFile))) {
+                    systemId = new BufferedReader(reader).lines().collect(Collectors.joining("\n"));
+                } catch (IOException e) {
+                    log.errorf(e, "/etc/machine-id exists and is readable, but exception was raised when reading it");
+                    systemId = "";
```
Comment (abonas) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/util/Util.java:326]: why is there a need to assign here and below an empty string? (what I mean is why not proceeding working with the null from the beginning of the method)

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java (hunk 2 above)
Comment (jpkrohling) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:531]: This is one of the syntax sugar changes I made while I was browsing the code. Most of those changes are not related and I intend to remove them, unless the Agent is meant to work on Java8+ only, in which case those changes are harmless.

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java [hunk 4]
```diff
@@ -455,12 +475,13 @@ private static void determineMetricSetPrometheus(ModelNode config,
 
         ModelNode platformValueNode = asPropertyList.get(0).getValue();
         boolean typeSetsEnabled = getBoolean(platformValueNode, context, PlatformAttributes.ENABLED);
-        if (typeSetsEnabled == false) {
+        if (!typeSetsEnabled) {
```
Comment (jpkrohling) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:478]: Unrelated change. Should I revert?

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java [hunk 5]
```diff
@@ -505,7 +537,7 @@ private static void determineMetricSetPrometheus(ModelNode config,
 
                     MetricType<PlatformNodeLocation> totalSpace = new MetricType<PlatformNodeLocation>(null,
                             Constants.FILE_STORE_TOTAL_SPACE,
-                            new AttributeLocation<PlatformNodeLocation>(
+                            new AttributeLocation<>(
```
Comment (jpkrohling) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:540]: Unrelated change. Should I revert?

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/protocol/platform/Constants.java [hunk 6]
```diff
@@ -42,7 +42,7 @@
                 Arrays.asList(MEMORY_AVAILABLE, MEMORY_TOTAL)),
 
         PROCESSOR("Processor",
-                Arrays.asList(PROCESSOR_CPU_USAGE)),
+                Collections.singletonList(PROCESSOR_CPU_USAGE)),
```
Comment (jpkrohling) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/protocol/platform/Constants.java:45]: Unrelated change. Should I revert?

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/storage/HawkularStorageAdapter.java [hunk 7]
```diff
@@ -42,7 +42,6 @@
 public class HawkularStorageAdapter implements StorageAdapter {
     private static final MsgLogger log = AgentLoggers.getLogger(HawkularStorageAdapter.class);
 
-    private String feedId;
```
Comment (jpkrohling) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/storage/HawkularStorageAdapter.java:45]: Unrelated change. Should I revert?

File: hawkular-wildfly-agent/src/test/java/org/hawkular/agent/monitor/protocol/platform/OshiPlatformCacheTest.java [hunk 8]
```diff
@@ -56,51 +56,37 @@ public void testConcurrency() {
         final CountDownLatch doneLatch1 = new CountDownLatch(1);
         final CountDownLatch doneLatch2 = new CountDownLatch(1);
 
-        Thread t1 = new Thread(new Runnable() {
```
Comment (jpkrohling) [inline hawkular-wildfly-agent/src/test/java/org/hawkular/agent/monitor/protocol/platform/OshiPlatformCacheTest.java:59]: Unrelated change. Should I revert?

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java (hunk 2 above)
Comment (jpkrohling) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:531]: I just marked the other unrelated changes, similar to this one.

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java (hunk 4 above)
Comment (abonas) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:478]: it's better to have PRs with code focused on a specific change that is advertised in the PR/commit message as a general best practice, however I'm not familiar with how PRs reviews work specifically in the agent repo.

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/util/Util.java (hunk 3 above)
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/util/Util.java:326]: if we keep it at null, it will keep attempting to read the file (and keep getting exceptions) everytime getSystemId() is called. This just forces systemId to be cached with a non-null value so we don't continue to try to obtain it when its just going to keep failing.

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java (hunk 5 above)
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:540]: I like using <> when allowed - makes the code cleaner. I'm fine with those changes to remove the generic type inside <>

File: hawkular-wildfly-agent/src/test/java/org/hawkular/agent/monitor/protocol/platform/OshiPlatformCacheTest.java (hunk 8 above)
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/test/java/org/hawkular/agent/monitor/protocol/platform/OshiPlatformCacheTest.java:59]: this is a test - no biggie. I would just leave everything as is. The other changes were minimal. I'm ok with leaving this PR as is if its working and passing tests.

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java (hunk 1 above)
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java:449]: this is how wildfy builds the model node data - I believe you get an error if you don't try to read the config element here as a property list.
